plugins/modules/vios.py

```diff
@@ -649,6 +731,81 @@ def installVios(module, params):
 
     return changed, vios_property, warn_msg
 
+def installViosUsingDisk(module, params):
+    hmc_host = params['hmc_host']
+    hmc_user = params['hmc_auth']['username']
+    password = params['hmc_auth']['password']
+    vios_iso = params['vios_iso']
+    image_dir = params['image_dir']
+    vios_IP = params['vios_IP']
+    vios_gateway = params['vios_gateway']
+    vios_subnetmask = params['vios_subnetmask']
+    network_macaddr = params['network_macaddr']
+    system_name= params['system_name']
+    name = params['name']
+    prof_name = params['prof_name'] or 'default_profile'
+    label = params['label']
+    timeout = params['timeout'] or 60
+    validate_parameters(params)
+    hmc_conn = HmcCliConnection(module, hmc_host, hmc_user, password)
+    hmc = Hmc(hmc_conn)
+    changed = False
+    vios_property = None
+    warn_msg = None
+    
+    if timeout < 10:
+        module.fail_json(msg="timeout should be more than 10mins")
+    try:
+        if network_macaddr:
+            hmc.installOSFromDisk(vios_iso, image_dir, vios_IP, vios_gateway, vios_subnetmask, network_macaddr, system_name, name, prof_name, label)
+        else:
+            dvcdictlt = hmc.fetchIODetailsForNetboot(hmc_host, vios_gateway, vios_IP, name, prof_name, system_name, vios_subnetmask)
+            for dvcdict in dvcdictlt:
+                if dvcdict['Ping Result'] == 'successful':
+                    network_macaddr = dvcdict['MAC Address']
+                    break
+            if network_macaddr:
+                hmc.installOSFromDisk(vios_iso, image_dir, vios_IP, vios_gateway, vios_subnetmask, network_macaddr, system_name, name, prof_name, label)
+            else:
+                module.fail_json(msg="Mac address not retrievable.") 
+
+        rmc_state, vios_property, ref_code = hmc.checkForOSToBootUpFully(system_name, name, timeout)
+        if rmc_state:
+            changed = True
+            warn_msg = "VIOS installation has been successfull"
+        elif ref_code in ['', '00']:
+            changed = True
+            warn_msg = "VIOS installation has been successfull but RMC didnt come up, please check the HMC firewall and security"
+        else:
+            module.fail_json(msg="VIOS Installation failed even after waiting for " + str(timeout) + " mins and the reference code is " + ref_code)
+
+    except HmcError as install_error:
+        return False, repr(install_error), None
+    
+    return changed, vios_property, warn_msg
```

Comment:
Does this vios_property contains os level as well

Reply:
No actually this contains the vios configuration, like at that point is it in running or in non activated state, but not the vios version.